src/snowflake/cli/plugins/streamlit/manager.py

```diff
@@ -112,7 +113,13 @@ def deploy(
                 query_warehouse=query_warehouse,
                 experimental=True,
             )
-            self._execute_query(f"ALTER streamlit {streamlit_name} CHECKOUT")
+            try:
+                self._execute_query(f"ALTER streamlit {streamlit_name} CHECKOUT")
+            except ProgrammingError as e:
+                # Handle case that CHECKOUT has already been created
+                if "Checkout already exists" not in str(e):
+                    raise
```

Comment:
You could consider creating dedicated Error ie. `StreamlitCheckoutMissing(ProgrammingError)`
and re-raising this error `StreamlitCheckoutMissing(e)`

It will result in more detailed context for error.


Also a test case for this behaviour would be useful.

Reply:
Sorry if this isn't very clear, but I don't think a custom exception is actually what I want in this case, because I want to:
1. Skip any "Checkout already exists" errors, because that's really not an issue that I care about when re-deploying a streamlit app, and otherwise the first `deploy` succeeds and subsequent ones fail
2. Any other errors should be raised as expected

Reply:
I switched to if/else as I think this makes it more clear now